.changeset/tasty-items-yell.md

```diff
@@ -0,0 +1,8 @@
+---
+"wrangler": minor
+---
+
+feat: Simplify `wrangler pages download config`:
+
+- If preview and production are currently equal, don't generate a production override
+- Don't include inheritable keys in the production override if they're equal to production
```

Comment:
I kind of felt like the behaviour should be:

* always create both preview and production environments
* create inheritable properties at the top level if they are the same for both environments
* add the inheritable properties in both the environments if they are not the same

Was this considered?

Reply:
the phrasing here is a bit confusing? I'm assuming "inheritable keys" refers to the top-level keys?

Reply:
also, to Pete's comment, and given the "incident" we had related to `production` defaulting to top level config if not defined, I was wondering if we need to revisit how `pages config download` should work. To my understanding, one of the things that came out of that incident is that we'd enforce declaring the `production` config as a means to introduce friction around users accidentally deploying dev config to prod. If that is the case, `pages config download` should do the same.

Reply:
After discussion, we're going to:
- always create the production environment
- create inheritable properties at the top level if they are the same for both environments
- add the inheritable properties in both the environments if they are not the same
- create the preview environment if needed